cmd/setec/setec.go

```diff
@@ -140,3 +212,154 @@ func runServer(ctx context.Context, args []string) error {
 
 	return nil
 }
+
+func clientFromArgs(args []string) (client *setec.Client, remainingArgs []string, err error) {
+	if len(args) < 1 {
+		return nil, nil, flag.ErrHelp
+	}
+	server := args[0]
+	// TODO(corp/13375): make a better UX here where single-label
+	// hostnames get expanded into a full HTTPS URL.
+	return &setec.Client{
+		Server: server,
+	}, args[1:], nil
+}
+
+func runList(ctx context.Context, args []string) error {
+	c, args, err := clientFromArgs(args)
+	if err != nil {
+		return err
+	}
+	if len(args) > 0 {
+		return flag.ErrHelp
+	}
+
+	secrets, err := c.List(ctx)
+	if err != nil {
+		return fmt.Errorf("Failed to list secrets: %v", err)
+	}
+
+	tw := tabwriter.NewWriter(os.Stdout, 0, 8, 0, '\t', 0)
+	io.WriteString(tw, "NAME\tACTIVE\tVERSIONS\n")
+	for _, s := range secrets {
+		vers := make([]string, 0, len(s.Versions))
+		for _, v := range s.Versions {
+			vers = append(vers, v.String())
+		}
+		fmt.Fprintf(tw, "%s\t%s\t%s\n", s.Name, s.ActiveVersion, strings.Join(vers, ","))
+	}
+	if err := tw.Flush(); err != nil {
```

Comment:
Done. This is a stylistic... choice? Habit? that I have, where sub-function calls in non-trivial functions come with their own error handling, and the function always gets a `return nil` epilogue of its own. I think I do this because that means inserting new code or modifying behavior doesn't require re-expanding the return statement into something else.

That said, it doesn't matter, and in this case it was definitely a bit much.